sandbox/lua/modules/ts_line_protocol.lua

```diff
@@ -148,23 +147,40 @@ local function points_tags_tables(config)
                 field = field:gsub("[^%w]", "_")
             end
 
-            -- Include the dynamic fields as tags if they are defined in
-            -- configuration or the magic value "**all**" is defined.
-            -- Convert value to a string as this is required by the API
-            if not config["carbon_format"]
-                and (config["tag_fields_all"]
-                or (config["used_tag_fields"] and used_tag_fields[field])) then
-                    table.insert(tags, influxdb_kv_fmt(field).."="..tostring(influxdb_kv_fmt(value)))
+            field_out_name = field
+
+            field_idx = seen_fields[field] or -1
+            field_idx = field_idx + 1
+            seen_fields[field] = field_idx
+            if field_idx > 0 then
+                field_out_name = field_out_name .. "_fidx_" .. tostring(field_idx)
```

Comment:
Prefer string.format for anything over a single concatenation (your call, I would have to profile it to see if it is worth it)
```lua
field_out_name = string.format("%s_fidx_%d", field_out_name, field_idx)
```